apps/geo/serializers.py

```diff
@@ -148,3 +148,38 @@ class Meta:
             'region_title', 'admin_level_level', 'admin_level_title',
             'parent'
         )
+
+
+class RegionGqSerializer(ProjectPropertySerializerMixin, UserResourceSerializer):
+    project = serializers.IntegerField()
+
+    class Meta:
+        model = Region
+        fields = '__all__'
+
+    def validate_project(self, project):
+        try:
+            project = Project.objects.get(id=project)
+        except Project.DoesNotExist:
+            raise serializers.ValidationError(
+                'Project matching query does not exist'
+            )
+
+        if not project.can_modify(self.context['request'].user):
+            raise serializers.ValidationError('Invalid project')
+        return project.id
+
+    def validate(self, data):
+        if self.instance and self.instance.is_published:
+            raise serializers.ValidationError('Published region can\'t be changed. Please contact Admin')
```

Comment:
```suggestion
            raise serializers.ValidationError('Published region can't be changed. Please contact Admin')
```

Reply:
@Rup-Narayan-Rajbanshi \ is to escape ' here.
https://www.w3schools.com/python/python_strings_escape.asp